File: velolabs_repos/lattis_android/app/src/main/java/com/lattis/ellipse/domain/interactor/updatetrip/StartLocationTrackInActiveTripUseCase.java
package com.lattis.ellipse.domain.interactor.updatetrip;

import com.lattis.ellipse.domain.executor.PostExecutionThread;
import com.lattis.ellipse.domain.executor.ThreadExecutor;
import com.lattis.ellipse.domain.interactor.UseCase;
import com.lattis.ellipse.presentation.model.LockModel;
import com.lattis.ellipse.presentation.ui.ride.service.util.ServiceAction;
import com.lattis.ellipse.presentation.ui.ride.service.util.UpdateTripData;

import javax.inject.Inject;

import io.reactivex.Observable;

public class StartLocationTrackInActiveTripUseCase extends UseCase<Boolean>

{
    private LockModel lockModel;
    private ServiceAction serviceAction;

    @Inject
    public StartLocationTrackInActiveTripUseCase(ThreadExecutor threadExecutor,
                                  PostExecutionThread postExecutionThread,
                                  ServiceAction serviceAction) {
        super(threadExecutor, postExecutionThread);
        this.serviceAction = serviceAction;
    }


    public StartLocationTrackInActiveTripUseCase withLockModel(LockModel lockModel) {
        this.lockModel = lockModel;
        return this;
    }

    @Override
    protected Observable<Boolean> buildUseCaseObservable() {
        return serviceAction.startLocationTracking(lockModel);
    }
}

File: velolabs_repos/lattis_android_v2/domain/src/main/java/com/lattis/domain/usecase/activetripservice/StartLocationTrackInActiveTripUseCase.kt
package com.lattis.domain.usecase.activetripservice


import com.lattis.domain.executor.PostExecutionThread
import com.lattis.domain.executor.ThreadExecutor
import com.lattis.domain.repository.ActiveTripRepository
import com.lattis.domain.usecase.base.UseCase
import com.lattis.domain.models.Lock
import javax.inject.Inject

import io.reactivex.rxjava3.core.Observable

class StartLocationTrackInActiveTripUseCase @Inject
constructor(threadExecutor: ThreadExecutor,
            postExecutionThread: PostExecutionThread,
            val activeTripRepository: ActiveTripRepository
) : UseCase<Boolean>(threadExecutor, postExecutionThread) {
    private var lock: Lock? = null


    fun withLock(lock: Lock): StartLocationTrackInActiveTripUseCase {
        this.lock = lock
        return this
    }

    override fun buildUseCaseObservable(): Observable<Boolean> {
        return activeTripRepository.startLocationTracking(lock!!)
    }
}
